src/main/app/case/case-api.ts

```diff
@@ -19,38 +28,45 @@ export class CaseApi {
   constructor(private readonly apiClient: CaseApiClient) {}
 
   public async getOrCreateCase(serviceType: DivorceOrDissolution, userDetails: UserDetails): Promise<CaseWithId> {
-    const userCase = await this.searchCases(serviceType);
-
-    return userCase || this.apiClient.createCase(serviceType, userDetails);
-  }
-
-  private async searchCases(serviceType: DivorceOrDissolution): Promise<CaseWithId | false> {
     if (config.get('services.case.checkDivCases') && (await this.hasInProgressDivorceCase())) {
       throw new InProgressDivorceCase('User has in progress divorce case');
     }
+    const userCase = await this.apiClient.getLatestLinkedCase(CASE_TYPE, serviceType);
 
-    const latestCase = (await this.apiClient.findUserCases(CASE_TYPE)).filter(
-      c => c.case_data.divorceOrDissolution === serviceType
-    )[0];
-
-    return latestCase
-      ? { ...fromApiFormat(latestCase.case_data), id: latestCase.id.toString(), state: latestCase.state }
-      : false;
+    return userCase || this.apiClient.createCase(serviceType, userDetails);
   }
 
   public async getCaseById(caseId: string): Promise<CaseWithId> {
     return this.apiClient.getCaseById(caseId);
   }
 
-  public async isAlreadyLinked(serviceType: DivorceOrDissolution, user: UserDetails): Promise<boolean> {
-    const userCase = await this.searchCases(serviceType);
+  public async isApplicantAlreadyLinked(serviceType: DivorceOrDissolution, user: UserDetails): Promise<boolean> {
+    const userCase = await this.apiClient.getLatestCaseOrInvite(CASE_TYPE, serviceType, user.email);
     if (userCase) {
       const userRoles = await this.apiClient.getCaseUserRoles(userCase.id, user.id);
-      return [UserRole.CREATOR, UserRole.APPLICANT_2].includes(userRoles.case_users[0]?.case_role);
+      const linkedRoles =
+        userCase.applicationType && userCase.applicationType.includes(ApplicationType.JOINT_APPLICATION)
+          ? [UserRole.CREATOR, UserRole.APPLICANT_2]
```

Comment:
we're checking whether the user is linked or not so that if they're linked we direct them straight to their existing application rather than the access-code page. the differentiation between joint and sole is because there are some joint cases where the second user created a case instead of linking to a case, and that case became the main case. but they still use the email link to login